examples/flow/src/flowEditor/components/LoadModal.tsx

```diff
@@ -0,0 +1,255 @@
+import { GraphJSON } from '@behave-graph/core';
+import { FC, useState, useEffect, useCallback, useMemo, CSSProperties } from 'react';
+import { useReactFlow } from 'reactflow';
+import { Modal } from './Modal';
+import { useDropzone } from 'react-dropzone';
+
+import { emptyGraphJson, fetchModelFile, publicUrl } from '../../hooks/useBehaveGraphFlow';
+import ModelPreview from '../../scene/ModelPreview';
+
+const modelFiles = {
+  courtyard: 'CourtYard.glb',
+  pressButtonToStartElevator: 'PressButtonToStartElevator.gltf',
+  suzanne: 'SpinningSuzanne.gltf',
+};
+
+const graphFiles = {
+  clickButtonToAnimate: 'ClickButtonToAnimate.json',
+  spinningSuzanne: 'SpinningSuzanne.json',
+  delay: 'Delay.json',
+  flipFlop: 'FlipFlop.json',
+  forLoop: 'FoorLoop.json',
+  helloWorld: 'HelloWorld.json'
+};
+
+export const examplePairs: ([string, string]|[string])[] = [
+  [modelFiles.pressButtonToStartElevator, graphFiles.clickButtonToAnimate],
+  [modelFiles.suzanne, graphFiles.spinningSuzanne],
+  [graphFiles.helloWorld],
+  [graphFiles.delay],
+  [graphFiles.flipFlop],
+  [graphFiles.forLoop],
+];
+
+const buildExampleOptions = () => examplePairs.map((pair, i) => {
+  if (pair.length === 2) {
+    const [modelFile, behaviorFile] = pair;
+    return {
+      index: i,
+      text: `Model: ${modelFile} / Behavior: ${behaviorFile}`,
+      modelFile,
+      behaviorFile,
+    }
+  } else {
+    const [behaviorFile] = pair;
+    return {
+      index: i,
+      text: `Behavior: ${behaviorFile}`,
+      behaviorFile,
+      modelFile: undefined
+    }
+  }
+
+  
+});
+
+export type LoadModalProps = {
+  open?: boolean;
+  onClose: () => void;
+  handleGraphJsonLoaded: (value: GraphJSON) => void;
+  setModelFile: (file: File|undefined) => void; 
+};
+
+const baseStyle = {
+  flex: 1,
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  padding: '20px',
+  borderWidth: 2,
+  borderRadius: 2,
+  borderColor: '#eeeeee',
+  borderStyle: 'dashed',
+  backgroundColor: '#fafafa',
+  color: '#bdbdbd',
+  outline: 'none',
+  transition: 'border .24s ease-in-out',
+};
+
+const focusedStyle = {
+  borderColor: '#2196f3',
+};
+
+const acceptStyle = {
+  borderColor: '#00e676',
+};
+
+const rejectStyle = {
+  borderColor: '#ff1744',
+};
+
+const useDropZoneStyle = ({
+  isFocused,
+  isDragAccept,
+  isDragReject,
+}: {
+  isFocused: boolean;
+  isDragAccept: boolean;
+  isDragReject: boolean;
+}) => {
+  const style = useMemo(
+    () => ({
+      ...baseStyle,
+      ...(isFocused ? focusedStyle : {}),
+      ...(isDragAccept ? acceptStyle : {}),
+      ...(isDragReject ? rejectStyle : {}),
+    }),
+    [isFocused, isDragAccept, isDragReject]
+  ) as CSSProperties;
+
+  return style;
+};
+
+export const LoadModal: FC<LoadModalProps> = ({ open = false, onClose, handleGraphJsonLoaded, setModelFile}) => {
```

Comment:
This component was modified to allow loading a 3d model, as well as preview it, but based on our discussion in the PR it probably makes more sense to have the 3d model loaded somewhere else, as to not clutter the core flow-editor with three.js related things.  I can work on reverting that change.